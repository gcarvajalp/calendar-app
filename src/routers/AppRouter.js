import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import CalendarScreen from '../components/calendar/CalendarScreen';
import AuthRouter from './AuthRouter';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';


const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute path="/auth" component={AuthRouter}  />
          <PrivateRoute path="/" component={CalendarScreen} />
        </Switch>
      </div>
    </Router>
  )
}

export default AppRouter;
