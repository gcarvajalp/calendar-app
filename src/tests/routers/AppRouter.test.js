import { mount } from 'enzyme';
import AppRouter from '../../routers/AppRouter';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

//store.dispatch = jest.fn();


describe('<AppRouter/>', () => {

  test('snapshot', () => {

    const initState = {
      auth: {
        checking: true
      }
    };

    const store = mockStore(initState);


    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h2').exists()).toBe(true);
  });

});
