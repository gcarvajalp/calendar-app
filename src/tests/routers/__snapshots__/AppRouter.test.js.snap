// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter/> snapshot 1`] = `
<h2>
  Espere...
</h2>
`;
